### PR TITLE
Follow redirects (302) while calling pre/post request handlers

### PR DESCRIPTION
**Pull request by damageboy**
- Disable the automatic handling of redirects (302) by the http stack
- Instead do it manually
- This results in the PreRequest/PostRequests handlers being called
  while handling the 302, which allows them to record cookies
  transmitted ONLY during the redirect response, as an example

### CODE REVIEW

**Comment by JonathanMagnan** [comment]
Hello @damageboy ,

Thank you for this pull, I will certainly check it this week.

Best Regards,

Jonathan

**Comment by damageboy** [comment]
Thanks.

I've used it successfully with the .NET Framework build, but have not used the Net Standard build with this fix...

**Comment by JonathanMagnan** [comment]
Hello @damageboy ,

I'm currently reviewing the change.

Unfortunately, I cannot change the AllowAutoRedirect to false. The risk to break a lot of libraries must be taken into this. I try to stay as much `backward compatible` as possible.

However, I can add the option to allow to set the `AllowAutoRedirect` to false by default and use the same logic as you provided when it `false`

Let me know if this solution could work for you.

Best Regards,

Jonathan

**Comment by damageboy** [comment]
I have no issue with doing this via an additional property (`AllowAutoRedirect` sounds good to me)
But I think there might be a mis-understanding here...

While on the surface of it, my patch does "disable" `AllowAutoRedirect` on the request, it does (later within the method) handle it "manually" (i.e. invokes `Get()` on the new redirected URL) and performs the redirect before returning the loaded document back to the user who called the API.

The reasoning for this is so that the Pre/PostRequest handlers can be called for the redirection as well....

I was encountering scraping situations where a critical cookie(s) were being returned only on the HTTP 302 response, and without this PR, those cookies would never be seen by the client, therefore, they would never be "recorded" and could not be provided later in future requests...

In short, I do believe that handling redirect the way I propose via this PR, is the right way to do so GENERALLY for this sort of a library, but I definitely understand if you would like to minimize any breakage we possibly haven't t thought of by making it conditional...

**Comment by JonathanMagnan** [comment]
Thank you, so perhaps finding another name for the option will be more appropriate.

Yup, unfortunately, some default behavior cannot be changed even if wrong since the impact will be to high.

Unless the option `BackwardCompatible` I recently added is turned to false.

**Comment by damageboy** [comment]
Sure thing, Maybe something like `CaptureRedirects` will do it then?

**Comment by damageboy** [comment]
No worries... Thanks for keeping me up to date...

I'm in the process of migrating to .net standard on a large project with
many dependencies so I'm not in a rush :)


    Shechter.

**Comment by JonathanMagnan** [comment]
Hello @ohadschn ,

The v1.6.4 has been released.

Let us know if everything works as expected.

Best Regards,

Jonathan

**Comment by ohadschn** [comment]
Hey @JonathanMagnan, not sure why you mentioned me?

**Comment by JonathanMagnan** [comment]
Hello @ohadschn ,

Sorry, my mistake. A bad copy & paste from my previous message. I forget to change the "@" name to @damageboy ;)

Best Regards,

Jonathan
